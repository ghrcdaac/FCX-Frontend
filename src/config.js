/*
    Following are the settings for FCX that should be configured per deployment environment.
    
    TODO:
    Don't freak out, the default value are left here only to make it easy to get started working with the code base. They are not guaranteed to work.
    In future, actual values of these variables should be directly read from environment variables and defaults should be removed.
*/

const dataBaseUrl = process.env.DATA_BASE_URL || "https://ghrc-fcx-viz-output.s3-us-west-2.amazonaws.com"
const abiBaseUrl = process.env.ABI_BASE_URL || "https://q6eymnfsd9.execute-api.us-west-2.amazonaws.com"
const flightTrackBaseUrl = process.env.FLIGHT_TRACK_BASE_URL || "https://fcx-czml.s3.amazonaws.com"
const mapboxAccessToken = process.env.MAPBOX_ACCESS_TOKEN || "pk.eyJ1IjoiYWppbmt5YWt1bGthcm5pIiwiYSI6ImNrYzY1NHh3bDA3cXIyenA4dDdpOGlja2UifQ.g4r3pjugLK8yC2mkQmJr4w"
const mapboxStyleId = process.env.MAPBOX_STYLE || "ckc65ettm0ixc1ipexrpklr96"
const mapboxUsername = process.env.MAPBOX_USERNAME || "ajinkyakulkarni"
const mapboxUrl = "https://api.mapbox.com/styles/v1/" + mapboxUsername + "/" + mapboxStyleId + "/tiles/256/{z}/{x}/{y}?access_token=" + mapboxAccessToken
const cesiumDefaultAccessToken =
  process.env.CESIUM_DEFAULT_TOKEN || "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJqdGkiOiIyNmQyZDgxNS0xNzY4LTQ4MzctOWZkOC02YmZkYjQ1ZGI1MDUiLCJpZCI6ODU1MSwic2NvcGVzIjpbImFzciIsImdjIl0sImlhdCI6MTU5NDM4OTIxMn0.WPkQXxpVgW5AfQdZKB15dMnLxQIjmA_Q2qimnDXONvc"
const supportEmail = process.env.GHRC_SUPPORT_EMAIL || "support-ghrc@earthdata.nasa.gov"

export { dataBaseUrl, abiBaseUrl, flightTrackBaseUrl, mapboxUrl, cesiumDefaultAccessToken, supportEmail }
