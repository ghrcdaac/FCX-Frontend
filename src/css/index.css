@import "../../node_modules/cesium/Build/Cesium/Widgets/widgets.css";

body {
  margin: 0;
  padding: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace;
}

a {
  text-decoration: none;
  color: #337ab7;
  background-color: transparent;
}

.th-wrapper {
  position: sticky;
}

.header {
  position: fixed;
}