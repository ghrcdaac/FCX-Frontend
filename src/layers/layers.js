import goes_r_plt_campaign from './goes-r-plt-layers'
import impacts_campaign from './impacts-layers'


const campaigns = {
  "goes-r-plt": goes_r_plt_campaign,
  // "impacts": impacts_campaign,
}

const getCampaignInfo = (mission) => {
  return campaigns[mission]
}

export default campaigns
export { getCampaignInfo }