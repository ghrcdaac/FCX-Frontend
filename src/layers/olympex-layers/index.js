const campaign = "olympex"
const olympex_campaign = {
    title: `${campaign} Field Campaign`,
    // logo: `${dataBaseUrl}/fieldcampaign/${campaign.toLowerCase()}/logo/${campaign.toLowerCase()}_logo_small.png`,
    logo: `missions-logos/impacts.png`,
    description: "",
    links: "",
    dois: "",
    layers: "",
    legends: "",
    defaultCamera: "",
  }
  
  export default olympex_campaign