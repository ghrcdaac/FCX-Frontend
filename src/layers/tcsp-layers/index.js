import { campaign, logo, description, dois, links, legends, defaultCamera, instrumentLayers } from "./components";

const tcsp_campaign = {
    title: `${campaign} Field Campaign`,
    logo: logo,
    description: description,
    links: links,
    dois: dois,
    legends: legends,
    defaultCamera: defaultCamera,
    layers: instrumentLayers
}

export default tcsp_campaign