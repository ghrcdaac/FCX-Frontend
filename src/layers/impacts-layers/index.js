import { dataBaseUrl } from "../../config"
import {
  cpl_er2,
  crs_er2,
  lip_er2,
  hiwrap_er2,
  defaultCamera,
  legends,
  links,
  dois,
  campaign,
  description
} from "./data"
import LayerGenerator from '../utils/LayerGenerator'

const generator = new LayerGenerator(campaign)

generator.addLayer({
  instrument: 'crs',
  platform: 'air-er2'
}, crs_er2)

generator.addLayer({
  instrument: 'lip',
  platform: 'air-er2'
}, lip_er2)

generator.addLayer({
  instrument: 'cpl',
  platform: 'air-er2'
}, cpl_er2)

generator.addLayer({
  instrument: 'hiwrap',
  platform: 'air-er2'
}, hiwrap_er2)

const layers = generator.generateLayers()

const impacts_campaign = {
  title: `${campaign} Field Campaign`,
  logo: `${dataBaseUrl}/fieldcampaign/${campaign.toLowerCase()}/logo/${campaign.toLowerCase()}_logo_small.png`,
  description,
  links,
  dois,
  layers,
  legends,
  defaultCamera,
}

export default impacts_campaign