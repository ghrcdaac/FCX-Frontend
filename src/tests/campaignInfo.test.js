import { shallow } from "enzyme";
import toJSON from "enzyme-to-json"
import React from "react";
import { getCampaignInfo } from '../layers/layers'

import '../customized-components/react-timeline-9000/src/setupTests'
import CampaignInfoLinks from "../components/campaignInfo";

describe("CampaignInfo Component", () => {
  it("Should Match CampaignInfo Snapshot", () => {
    const wrapper = shallow(
      <CampaignInfoLinks
        campaign={getCampaignInfo('goes-r-plt')}
      />
    )
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})