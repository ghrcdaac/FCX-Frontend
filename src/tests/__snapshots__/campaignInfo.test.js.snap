// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignInfo Component Should Match CampaignInfo Snapshot 1`] = `
Array [
  <div
    key="campaignInfo-0"
  >
    <FiExternalLink />
     
    <a
      href="https://ghrc.nsstc.nasa.gov/uso/ds_details/collections/goesrpltC.html"
      rel="noopener noreferrer"
      target="_blank"
    >
      Landing Page
    </a>
  </div>,
  <div
    key="campaignInfo-1"
  >
    <FiExternalLink />
     
    <a
      href="https://ghrc.nsstc.nasa.gov/home/micro-articles/goes-r-post-launch-test-plt"
      rel="noopener noreferrer"
      target="_blank"
    >
      Micro Article
    </a>
  </div>,
  <div
    key="campaignInfo-2"
  >
    <FiExternalLink />
     
    <a
      href="https://ghrc.nsstc.nasa.gov/home/field-campaigns/goes-r_plt"
      rel="noopener noreferrer"
      target="_blank"
    >
      Description
    </a>
  </div>,
  <div
    key="campaignInfo-3"
  >
    <FiExternalLink />
     
    <a
      href="https://ghrc.nsstc.nasa.gov/home/field-campaigns/goes-r_plt/instruments"
      rel="noopener noreferrer"
      target="_blank"
    >
      Instruments
    </a>
  </div>,
]
`;
